.github/workflows/publish_microsite.yml

```diff
@@ -0,0 +1,32 @@
+name: Publish Doc
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - docs/**
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Jekyll
+        run: |
+          gem install --install-dir $HOME/vendor/bundle jekyll
+          echo "GEM_HOME=$HOME/vendor/bundle" >> $GITHUB_ENV
+          echo "GEM_PATH=$HOME/vendor/bundle" >> $GITHUB_ENV
+          echo "$HOME/vendor/bundle/bin" >> $GITHUB_PATH
+      - name: Build docs
+        run: |
+          sbt docs/makeMicrosite
```

Comment:
Where does `sbt` come from? I think you usually need to install it with an action.

Instead of as a hand-written workflow, Do you think we can include this in the sbt build with sbt-gh-actions? Then most of the setup and compiling is done automatically. See e.g. https://github.com/typelevel/fs2/blob/main/build.sbt#L349

Thanks!

Reply:
>Where does sbt come from?

ubuntu-latest has sbt installed.

Reply:
@armanbilge 

I choose a hand-written workflow because I have one concern.  Is it possible for sbt-gh-actions to define multiple workflow? 

It is faster to publish docs when we use multiple workflows because workflow runs in parallel but steps do not run in parallel.

Reply:
@armanbilge 

> Do you think we can include this in the sbt build

Answer is "yes, we can" but do we really need that?

I think hand-written workflow is better when it comes to managing spire document because

- current publish task has little to do with Scala(only `mdoc` task), it is done by `jekyll` and `actions-gh-pages`. Wrapping them in sbt task does not seem straightforward.
- I took a look `sbt-gh-actions` and `sbt-gh-actions` focus mainly on making it easier to test, build and publish Scala artifacts **over cross java/scala versions and cross target platforms(jvm/native/js)** (though it is possible to add publish docs jobs).
  - In addition, if we include publish task  in jobs, we have to introduce another action to detect file changes, which can be done by github workflow's path filter.
- some libraries in typelevel repo like `cats-effect` uses hand-written workflow for publishing docs. https://github.com/typelevel/cats-effect/blob/docs/.github/workflows/microsite.yml
- separating document settings from build settings will keep build settings simple(and makes it easier to migrate to another document tools in the future.)

You say "setup and compiling is done automatically" but these features make little difference for publishing docs. We do not have to setup Java or Scala for publishing docs. We need setup `jekyll`, envs and paths even if we use `sbt-gh-actions`. 

If we publish docs for each spire versions over each Scala versions and each platforms it is worth managing it in build.sbt, but current microsite is independent of Scala versions or platforms, just displaying information of the latest version, so the publish doc workflow is simple enough and will not be likely to swell further.

Reply:
Ah, you mean you want to programatically garantee docs are successfully built using 'if ....' or 'needs build' ? If so, it would be better to include  the doc job in build.sbt.

Sorry, I assumed that docs are supposed to be successfully built everytime PR is merged to main (or version tag is set) because current workflow contains build docs step and you can be sure docs are vaild.

> So we should actually be careful to only publish the site when we release a new, stable version

Ok, workflow becomes simpler and there's no need to check doc file changes as publishing documents happens only when stable release, not when doc files changes.

Reply:
Yes, just for my own clarification the docs should be _built_ as part of every PR, to avoid breaking something. And as you point out, they already are. But of course, they should not be _published_ from the PR :)

> Ok, workflow becomes simpler and there's no need to check doc file changes as publishing documents happens only when stable release, not when doc files changes.

The downside to this is that docs contributions/updates are not published until the next release ... but, it prevents the API docs from getting out of sync with the published artifact, which could be very confusing. I'm open to ideas to improve this.

One way would be to publish the API docs under a version directory (e.g., 0.17, 0.18, SNAPSHOT). So this way, even though updating the site would also update the API docs, it would publish under SNAPSHOT instead of overwriting the API docs for the current.

For clarification, I'm distinguishing between the site/main docs (which should ideally be updated as frequently as possible) and the API docs aka scaladocs, which should match the published artifact.

Thoughts? Thanks again for your help with this, much appreciated!